src/PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -598,10 +602,11 @@ public int OnCmdUIContextChanged(uint dwCmdUICookie, int fActive)
             {
                 ThreadHelper.ThrowIfNotOnUIThread();
                 OnSolutionExistsAndFullyLoaded();
+
                 // We must call DeleteMarkedPackageDirectories outside of OnSolutionOpened, because OnSolutionOpened might be called in the constructor
                 // and DeleteOnRestartManager requires VsFileSystemProvider and RepositorySetings which both have dependencies on SolutionManager.
                 // In practice, this code gets executed even when a solution is opened directly during Visual Studio startup.
```

Comment:
I am not sure of the impact of not doing it here. The comment can ofcourse be deleted.